src/xcode/ENA/ENA/Source/Services/PPAnalyticsCollector/PPAnalyticsCollector.swift

```diff
@@ -0,0 +1,475 @@
+	static func setup(
+		store: Store,
+		submitter: PPAnalyticsSubmitter
+	) {
+		// Make sure the secure store now also implements the PPAnalyticsData protocol with the properties defined there (the analytics data proporties).
+		guard let store = store as? (Store & PPAnalyticsData) else {
+			Log.error("I will never submit any analytics data. Could not cast to correct store protocol", log: .ppa)
+			fatalError("I will never submit any analytics data. Could not cast to correct store protocol")
+		}
+		PPAnalyticsCollector.store = store
+		PPAnalyticsCollector.submitter = submitter
+	}
+	
+	#if !RELEASE
+
+	/// Setup Analytics for testing. The store or the submitter can be nil for testing purposes.
+	static func setupMock(
+		store: (Store & PPAnalyticsData)? = nil,
+		submitter: PPAnalyticsSubmitter? = nil
+	) {
+		PPAnalyticsCollector.store = store
+		PPAnalyticsCollector.submitter = submitter
+	}
+	
+	#endif
+
+	/// The main purpose for the collector. Call this method to log some analytics data and pass the corresponding enums.
+	static func collect(_ dataType: PPADataType) {
+
+		// Make sure the user consent is given. If not, we must not log something.
+		guard let consent = store?.isPrivacyPreservingAnalyticsConsentGiven,
+			  consent == true else {
+			Log.info("Forbidden to log any analytics data due to missing user consent", log: .ppa)
+			return
+		}
+
+		Log.debug("Logging analytics data: \(dataType)", log: .ppa)
+		switch dataType {
+		case let .userData(userMetadata):
+			Analytics.logUserMetadata(userMetadata)
+		case let .riskExposureMetadata(riskExposureMetadata):
+			Analytics.logRiskExposureMetadata(riskExposureMetadata)
+		case let .clientMetadata(clientMetadata):
+			Analytics.logClientMetadata(clientMetadata)
+		case let .testResultMetadata(TestResultMetadata):
+			Analytics.logTestResultMetadata(TestResultMetadata)
+		case let .keySubmissionMetadata(keySubmissionMetadata):
+			Analytics.logKeySubmissionMetadata(keySubmissionMetadata)
+		case let .exposureWindowsMetadata(exposureWindowsMetadata):
+			Analytics.logExposureWindowsMetadata(exposureWindowsMetadata)
+		case let .submissionMetadata(submissionMetadata):
+			Analytics.logSubmissionMetadata(submissionMetadata)
+		}
+
+		// At the end, try to submit the data. In the submitter are all the checks that we do not submit the data to often.
+		Analytics.triggerAnalyticsSubmission()
+	}
+
+	/// This removes all stored analytics data that we collected.
+	static func deleteAnalyticsData() {
+		store?.currentRiskExposureMetadata = nil
+		store?.previousRiskExposureMetadata = nil
+		store?.userMetadata = nil
+		store?.lastSubmittedPPAData = nil
+		store?.lastAppReset = nil
+		store?.lastSubmissionAnalytics = nil
+		store?.clientMetadata = nil
+		store?.testResultMetadata = nil
+		store?.keySubmissionMetadata = nil
+		store?.exposureWindowsMetadata = nil
+		Log.info("Deleted all analytics data in the store", log: .ppa)
```

Comment:
Having a dedicated analytics table that could be dropped completely would be nicer. I guess there is a reason why that solution didn't made it into the code?

Reply:
Yes, we dropped that idea because the effort got bigger and bigger and somewhere we wanted to do a cut.
The only functional reason now for complete new table is that noone forgets to add here "his" new property. For this, I wrote a test that secures that.
Benefit of not having a separated table: No need to setup and then migrate the database values from one to the other.

Reply:
Yes. It has a reason behind it. It based on difference criteria to empty the specific store as per tech specs.

Reply:
If the specs are over-specific we should discuss it in future. Imho, specs should provide the constraints in which we decide how to implement it…